### PR TITLE
Renames in serializer to reflect recent changes

### PR DESCRIPTION
**Pull request by rsgowman**
* Rename Writer::Encode* methods to Writer::Write*

* Rename: s/stream/writer/ (approximately)
but only where it applies to Writer (rather than pb_ostream_t).

### CODE REVIEW

File: Firestore/core/src/firebase/firestore/remote/serializer.cc [hunk 1]
```diff
@@ -455,37 +455,37 @@ FieldValue DecodeNestedFieldValue(pb_istream_t* stream) {
...
- * would be encoded (in proto text format) as:
+ * would be written (in proto text format) as:
  *   {
  *     fields: {key:"key string 1", value:{<Value message here>}}
  *     fields: {key:"key string 2", value:{<Value message here>}}
  *     ...
  *   }
  *
- * This method encodes an individual entry from that list. It is expected that
+ * This method writese an individual entry from that list. It is expected that
```
Comment (var-const) [inline Firestore/core/src/firebase/firestore/remote/serializer.cc]: Nit: s/writese/writes

Comment (rsgowman) [reply Firestore/core/src/firebase/firestore/remote/serializer.cc]: done.

File: Firestore/core/src/firebase/firestore/remote/serializer.cc [hunk 2]
```diff
@@ -565,10 +565,10 @@ std::map<std::string, FieldValue> DecodeObject(pb_istream_t* stream) {
 
 }  // namespace
 
-void Serializer::EncodeFieldValue(const FieldValue& field_value,
-                                  std::vector<uint8_t>* out_bytes) {
-  Writer stream = Writer::Wrap(out_bytes);
-  EncodeFieldValueImpl(&stream, field_value);
+void Serializer::WriteFieldValue(const FieldValue& field_value,
```
Comment (wilhuff) [inline Firestore/core/src/firebase/firestore/remote/serializer.cc]: The verbs on the Serializer are Encode/Decode

cf Android: http://google3/java/com/google/android/gmscore/dev/thick_client/firebase-firestore/src/com/google/firebase/firestore/remote/RemoteSerializer.java?rcl=185935790

Basically within the context here I think we should end up with Writer writes and Serializer encodes. The free functions in here seem like they're private members of the serializer so I think they're "encode" too.

Comment (rsgowman) [reply Firestore/core/src/firebase/firestore/remote/serializer.cc]: Whoops. Yeah, that's why I used 'Encode' in the first place. (Apparently, I got a little carried away.) Fixed.

(And yeah, I agree that the free functions are also "encode".)

**Pull request merged**